controllers/user_keys_test.go

```diff
@@ -288,4 +290,72 @@ func (suite *UserKeysTestSuite) TestUpdateExistingKey() {
 	suite.Equal(requestBody.Service, responseKey.Service)
 	suite.Equal(requestBody.KeyName, responseKey.KeyName)
 	suite.Equal(requestBody.KeyMaterial, responseKey.KeyMaterial)
+	suite.Equal(2, responseKey.SerialNumber) // Should be 2 since key was updated
+}
+
+func (suite *UserKeysTestSuite) TestSerialNumberIncrement() {
```

Comment:
optional nit: we could test to ensure that serial number sequences remain isolated between keys, either in this test or in another test

Reply:
Do you mean something like set `wrapping_key` for `accounts` (expecting a serial number of 2) and then set `enc_key` for `sync` (expecting a serial number of 1)?

Reply:
yes, something like that

Reply:
Added.